Sprint 09

bakgeldia opened the pull request:
Задача 9-спринта:
1. Экран профиля пользователя свёрстан кодом
2. Добавлен экран отдельного изображения SingleImageViewController
3. На экране SingleImageViewController реализована кнопка "Поделиться".

bakgeldia committed:
Сверстан экран профиля пользователя ProfileViewController в Main.storyboard согласно дизайну макета.

bakgeldia committed:
- Добавлен UITabBarController
- К нему подключены ImagesListViewController и ProfileViewController
- Добавлены иконки в TabBar

bakgeldia committed:
- Добавлен NavigationController
- Сконфигурирован внешний вид Navigation Bar

bakgeldia committed:
Добавлен SingleImageViewController

bakgeldia committed:
Переход с использованием Navigation Controller Push

bakgeldia committed:
Вместо Navigation Controller был использован показ модального окна, используя сегвей Present

bakgeldia committed:
- Добавлена функциональность зума картинки с помощью ScrollView.
- Добавлена кнопка "Поделиться" на экране SingleImageViewController

bakgeldia committed:
Экран пользователя сверстан кодом

helecta commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,98 @@
+    @IBOutlet private var logoutButton: UIButton!
+    
+    
+    var labelName: UILabel?
```
Стоит организовывать код следующим образом:

- Публичные переменные
- IBOutlet
- Приватные переменные
- Публичные методы(сначала `init`ы, `override`ы, и потом остальные)
- Приватные методы
- IBAction

Подробнее почитать можно тут: [Организация кода](https://github.com/tutu-ru/swift-style-guide?tab=readme-ov-file#Организация-кода)

helecta commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        self.labelDescription = description
+        
+        let button = UIButton.systemButton(
+            with: UIImage(named: "logout")!,
```
Даже в таких местах не следует использовать force unwrap, лучше избегать force операции. Т.к они могут приводить к падению приложения, считается что падение приложения хуже, чем не рабочее состояние. Нерабочее состояние можно обработать, сохранить последнее состояние приложения, что бы позже пользователь мог продолжить работу с того места на котором остановился. С падением уже ничего сделать нельзя, обычно это вызывает негатив у пользователей.

helecta replied:
В assets у тебя название изображения "Logout", поэтому приложение у меня падает

helecta replied:
Аналогичная проблема для кнопки назад, в сторборде она у тебя с маленькой буквы, в assets с большой

bakgeldia replied:
Странно, у меня они с везде с маленькой буквы. До этого были с большой, но я изменил. Почему-то с коммитом это не изменилось в гитхабе.

helecta replied:
я так и подумала, что проблема с коммитом возникла, поэтому не стала выносить это в крит замечание

helecta commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        self.labelUsername = username
+        
+        let description = UILabel()
+        description.text = "Hello, World!"
```
Это поле может быть длинным, в этом случае у тебя текст будет обрезаться.
Лучше тут сделать многострочный вариант строки. Для этого нужно сделать так:
```
description.numberOfLines = 0
```
В этом случае еще нужно добавить trailingAnchor для description

helecta commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        username.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(username)
+        username.leadingAnchor.constraint(equalTo: name.leadingAnchor).isActive = true
+        username.topAnchor.constraint(equalTo: name.bottomAnchor, constant: 8).isActive = true
```
Тут тоже лучше добавить trailingAnchor, чтобы если имя будет слишком длинным, то оно нормально обрезалось

bakgeldia replied:
Наверное вы имели ввиду про name

helecta replied:
и для nameLabel и для loginNameLabel

bakgeldia committed:
Исправления:
- Кнопка "поделиться" вынесена наружу со ScrollView
- Добавлен trailingAnchor для name, username, description

bakgeldia committed:
Изменен цвет кнопки выхода

bakgeldia committed:
Констрейнт для кнопки поделиться изменен с 17 на 16 (по дизайну в figma)
